pkg/common/cns-lib/volume/manager.go

```diff
@@ -2369,12 +2371,32 @@ func (m *defaultManager) createSnapshotWithImprovedIdempotencyCheck(ctx context.
 				log.Infof("Snapshot with name %q and id %q on Volume %q is already created on CNS with opId: %q.",
 					instanceName, volumeOperationDetails.SnapshotID, volumeOperationDetails.VolumeID,
 					volumeOperationDetails.OperationDetails.OpID)
-
+				log.Infof("Retriving aggregated Snapshot Capacity for volume with volumeID %q", volumeID)
+				aggregatedSnapshotCapacity, err := m.getAggregatedSnapshotSize(ctx, volumeID)
+				if err != nil {
+					return nil, logger.LogNewErrorf(log, "Failed to retrive volume with volumeID %q with error %v", volumeID, err)
+				}
+				taskMoRef := vim25types.ManagedObjectReference{
+					Type:  "Task",
+					Value: volumeOperationDetails.OperationDetails.TaskID,
+				}
+				createSnapshotsTask = object.NewTask(m.virtualCenter.Client.Client, taskMoRef)
+				createSnapshotsTaskInfo, err := cns.GetTaskInfo(ctx, createSnapshotsTask)
+				snapshotLatestOperationCompleteTime := *createSnapshotsTaskInfo.CompleteTime
+				if err != nil {
+					log.Warnf("Failed to retrive taskinfo for snapshot %q taskid %q and volumeID %q with error %v",
+						volumeOperationDetails.SnapshotID, volumeOperationDetails.OperationDetails.TaskID, volumeID, err)
+					log.Infof("Using create snapshot timestamp instead of create snapshot task completion time for snapshot %q and volumeID %q",
+						volumeOperationDetails.SnapshotID, volumeID)
+					snapshotLatestOperationCompleteTime = volumeOperationDetails.OperationDetails.TaskInvocationTimestamp.Time
```

Comment:
This does not seem correct, in this case you should have all the necessary info in the task results. Ideally, volumeOperationDetails should already contain this information. I think it'll look a lot cleaner if we introduce a new AggregatedSnapshotSize field in VolumeOperationDetails:

https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/pkg/internalapis/cnsvolumeoperationrequest/util.go#L55-L60
```go
type QuotaDetails struct {
	Reserved         *resource.Quantity
	StoragePolicyId  string
	StorageClassName string
	Namespace        string
        AggregatedSnapshotSize *resource.Quantity
}
```
This can be set after a task success. When this code is invoked, we can get all the details from VolumeOperationDetails.